orttraining/orttraining/test/gradient/gradient_ops_test.cc

```diff
@@ -667,98 +667,54 @@ TEST(GradientCheckerTest, ConvGrad) {
-  {
-    TensorShape x1_shape({2, 2});
-    TensorShape x2_shape({2, 4});
-    TensorShape y_shape({2, 6});
-    gradient_checker.ComputeGradientError(op_def, {x1_shape, x2_shape}, {y_shape}, &max_error,
-                                          {MakeAttribute("axis", int64_t(-1))}, true, false);
-    EXPECT_IS_TINY(max_error);
-  }
-}
-
-TEST(GradientCheckerTest, ConcatTrainingGrad) { /*also test w/o shape inferencing */
-  float max_error;
-  GradientChecker<float, float, float> gradient_checker;
-  OpDef op_def{"ConcatTraining", kMSDomain, 1};
+  bool extra_input = op_type == "ConcatTraining" ? true : false;
+  OpDef op_def{op_type, domain, opset_version};
 
   //concat_1d
   {
     TensorShape x_shape({2});
     TensorShape y_shape({6});
-    TensorShape y1_shape({3});
-    TensorInfo y1 = TensorInfo(y1_shape, false, nullptr, DataTypeImpl::GetTensorType<int64_t>());
-
+    TensorInfo y1 = TensorInfo({3}, false, nullptr, DataTypeImpl::GetTensorType<int64_t>());
+    std::vector<TensorInfo> output = {y_shape};
+    if(extra_input) output.push_back(y1);
     gradient_checker.ComputeGradientError(op_def, {x_shape, x_shape, x_shape},
-                                          {y_shape, y1}, &max_error,
-                                          {MakeAttribute("axis", int64_t(0))}, true, true);
+                                                                        output, &max_error,
+                                                                        {MakeAttribute("axis", int64_t(0))}, true,
+                                                                        check_not_have_shape_inferencing);
     EXPECT_IS_TINY(max_error);
   }
 
   //concat_2d
   {
     TensorShape x_shape({2, 2});
     TensorShape y_shape({2, 6});
-    TensorShape y1_shape({3});
-    TensorInfo y1 = TensorInfo(y1_shape, false, nullptr, DataTypeImpl::GetTensorType<int64_t>());
+    TensorInfo y1 = TensorInfo({3}, false, nullptr, DataTypeImpl::GetTensorType<int64_t>());
+    std::vector<TensorInfo> output = {y_shape};
+    if (extra_input) output.push_back(y1);
     gradient_checker.ComputeGradientError(op_def, {x_shape, x_shape, x_shape},
-                                          {y_shape, y1}, &max_error,
-                                          {MakeAttribute("axis", int64_t(1))}, true, true);
+                                          output, &max_error,
+                                          {MakeAttribute("axis", int64_t(1))}, true, 
+                                          check_not_have_shape_inferencing);
     EXPECT_IS_TINY(max_error);
   }
 
   //concat_3d
   {
     TensorShape x_shape({1, 2, 3});
     TensorShape y_shape({1, 2, 9});
-    TensorShape y1_shape({3});
-    TensorInfo y1 = TensorInfo(y1_shape, false, nullptr, DataTypeImpl::GetTensorType<int64_t>());
+    TensorInfo y1 = TensorInfo({3}, false, nullptr, DataTypeImpl::GetTensorType<int64_t>());
+    std::vector<TensorInfo> output = {y_shape};
+    if(extra_input) output.push_back(y1);
```

Comment:
scope of y1 could be limited to the conditional block. nit: whitespace
this comment applies to a few other places too